src/client/metadataApiDeploy.ts

```diff
@@ -348,11 +349,17 @@ export class MetadataApiDeploy extends MetadataTransfer<MetadataApiDeployStatus,
   }
 
   protected async pre(): Promise<AsyncResult> {
-    const [zipBuffer, connection] = await Promise.all([
-      this.getZipBuffer(),
-      this.getConnection(),
-      this.maybeSaveTempDirectory('metadata'),
-    ]);
+    const connection = await this.getConnection();
+    // store for use in the scopedPostDeploy event
+    this.orgId = connection.getAuthInfoFields().orgId;
+    // only do event hooks if source, (NOT a metadata format) deploy
+    if (this.options.components) {
+      await Lifecycle.getInstance().emit('scopedPreDeploy', {
+        componentSet: this.options.components,
+        orgId: this.orgId,
+      } as ScopedPreDeploy);
```

Comment:
Is this meant to replace the existing predeploy hooks in plugin-source or will they coincide for the life of sfdx?  Is this an internal only hook or can other plugins listen?

Reply:
Mike and I left that question TBD. We could either make these the official hooks, or maybe even emit the current sfdx hooks from SDR too.

There's not really a sense of "privileged" hooks, so there's nothing to stop people from seeing and using them.